pegasus/routes/dev_routes.rb

```diff
@@ -0,0 +1,27 @@
+BUILD_STARTED_PATH = deploy_dir('build-started')
+
+# Used to restart builds on staging/test via Slack /start-build slash command.
+post '/api/dev/start-build' do
+  # Forbidden in production because it's a dev route
+  # Forbidden in development because it won't work anyway
+  forbidden! if [:development, :production].include? rack_env
+
+  dont_cache
+
+  forbidden! unless params[:token] == CDO.slack_start_build_token
+
+  # Don't create build-started if it already exists; let the requester know.
+  if File.file?(BUILD_STARTED_PATH)
+    return "I can't do that #{params[:user_name]} - a build is already queued"
```

Comment:
This plaintext response will go back to the requesting user, but not to anyone else in the room ([docs](https://api.slack.com/slash-commands#responding_to_a_command)).  That seems appropriate for this particular situation, where the user will get what they want but their request is redundant with another.